pkg/platform/runtime/setup/setup.go

```diff
@@ -182,17 +199,12 @@ func (s *Setup) update() error {
 	changedArtifacts := artifact.NewArtifactChangesetByRecipe(oldRecipe, buildResult.Recipe, false)
 	s.events.ChangeSummary(artifacts, requestedArtifacts, changedArtifacts)
 
-	setup, err := s.selectSetupImplementation(buildResult.BuildEngine, artifacts)
-	if err != nil {
-		return errs.Wrap(err, "Failed to select setup implementation")
-	}
-
 	storedArtifacts, err := s.store.Artifacts()
 	if err != nil {
 		return locale.WrapError(err, "err_stored_artifacts", "Could not unmarshal stored artifacts, your install may be corrupted.")
 	}
 
-	alreadyInstalled := setup.ReusableArtifacts(changedArtifacts, storedArtifacts)
+	alreadyInstalled := reusableArtifacts(buildResult.BuildStatusResponse.Artifacts, storedArtifacts)
```

Comment:
This addresses a problem where the `alreadyInstalled` slice included too many artifacts.  They sneaked in there after a failed build was incompletely installed and reverted.  The artifacts that had been installed as part of the failed commit were never removed and therefore appeared incorrectly in that list.

The solution proposed in this PR, is simpler as it is  the same for camel and alternative builds) by just comparing the `storedArtifacts` with the list of the artifacts that the head-chef returns in its build status response.

Reply:
Follow-up story is here: https://www.pivotaltracker.com/story/show/178901061